include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_histogram.h

```diff
@@ -0,0 +1,547 @@
+// -*- C++ -*-
+//===-- parallel_backend_sycl_reduce.h --------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_PARALLEL_BACKEND_SYCL_HISTOGRAM_H
+#define _ONEDPL_PARALLEL_BACKEND_SYCL_HISTOGRAM_H
+
+#include <algorithm>
+#include <cstdint>
+#include <type_traits>
+
+#include "sycl_defs.h"
+#include "parallel_backend_sycl_utils.h"
+#include "execution_sycl_defs.h"
+#include "unseq_backend_sycl.h"
+#include "utils_ranges_sycl.h"
+#include "oneapi/dpl/internal/histogram_binhash_utils.h"
+#include "oneapi/dpl/internal/async_impl/async_impl_hetero.h"
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace __par_backend_hetero
+{
+
+//no boost baseline
+template <typename _BinHash>
+struct __binhash_SLM_wrapper
+{
+    _BinHash __bin_hash;
+    __binhash_SLM_wrapper(_BinHash __bin_hash) : __bin_hash(__bin_hash) {}
+
+    template <typename _T2>
+    inline ::std::uint32_t
+    get_bin(_T2&& __value) const
+    {
+        return __bin_hash.get_bin(::std::forward<_T2>(__value));
+    }
+
+    template <typename _T2>
+    inline bool
+    is_valid(_T2&& __value) const
+    {
+        return __bin_hash.is_valid(::std::forward<_T2>(__value));
+    }
+
+    inline ::std::size_t
+    get_required_SLM_memory() const
+    {
+        return 0;
+    }
+
+    inline void
+    init_SLM_memory(void* __boost_mem, const sycl::nd_item<1>& __self_item) const
+    {
+    }
+
+    template <typename _T2>
+    inline ::std::uint32_t
+    get_bin(_T2&& __value, void* __boost_mem) const
```

Comment:
`__boost_mem` is unused variable

Reply:
This is to provide a unified interface with the other specialization so that the bin hash can be treated generically in the kernel.  Evenly divided bins do not take advantage of extra SLM memory, but custom range bins do, to store the `__boundaries`.

Reply:
Would it be worth it to decorate `__boost_mem` as `[[maybe_unused]]`, so the user does not receive any additional warnings when compiling with `-Wunused-parameter`?

Reply:
Considering it further, it makes more sense to make it an unnamed parameter for this overload.  I think this makes it clear that it is intentionally unused and avoids those warnings as well.